pkg/dns/helpers.go

```diff
@@ -16,34 +16,20 @@ import (
 // ToRecordRequestsAndHash creates DNS records based on the ingress and domain supplied. It also returns a hash of those
 // records, suitable for using overtime to determine if the ingress's records need to change.
 func ToRecordRequestsAndHash(domain string, ingress *v1.Ingress) ([]RecordRequest, string) {
-	var ips, lbHosts, recordValues []string
+	var ipv4s, ipv6s, lbHosts, recordValues []string
 
 	for _, i := range ingress.Status.LoadBalancer.Ingress {
 		if i.IP != "" {
-			ips = append(ips, i.IP)
+			if strings.Contains(i.IP, ":") {
```

Comment:
I'd probable do net.ParseIP and check the result if it's v4 or v6. Maybe looking for : is sufficient, I don't know, but everything about IPv6 is terrible.

Reply:
I went around on this several times. Googled around a good bit. Surprising that there is not 1 clean and standard way.

My logic here was that this was the simplest solution give the constraints. The biggest constraint/assumption is that this field is already a validated IP address, so it will always be a basic ipv4 of v6 addr.

Are you strongly opposed to keeping it this way until proven wrong?

Reply:
I really think they only way it wouldnt be sufficient is if the addr contained a port and I don't think they will in this case. Interestingly, ParseIP decides which it is (and how to parse it) by which occurs first, a dot or colon